### PR TITLE
Add vim plugin coc.nvim segments.

### LINKED ISSUE #2056: Add vim plugin coc.nvim status segment
May we could think of supporting display vim plugin coc.nvim compile status (Error number and Warning number), like vim ale segment do.

### PR DESCRIPTION
**Pull request by chrisniael**
Close #2056

### CODE REVIEW

**Comment by PH111P** [comment]
Hi @chrisniael,

Nice work. Just a small thing: The tests seem to fail as you didn't add default colors for the new highlight groups defined in your segment:
```
Error while checking theme (key /segments/left/list item 12/function)
found highlight group coc not defined in the following colorschemes: default, solarized, solarizedlight
```

I'd be nice if you could fix this.

File: powerline/segments/vim/plugin/coc.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	for item in status_list:
+		if len(item) > 0 and item[0] == 'E':
+			errors_count = int(item[1:])
+		if len(item) > 0 and item[0] == 'W':
+			warnings_count = int(item[1:])
+	return errors_count, warnings_count
+
+@requires_segment_info
+def coc(segment_info, pl):
+	segments = []
```
Comment (PH111P) [inline powerline/segments/vim/plugin/coc.py]: The documentation for this function is missing: `powerline-lint` does not know how the highlight groups are named. Try adding:
```
    '''<add description here>

    Highlight groups used: ``coc:warning`` or ``warning``, ``coc:error`` or ``error``.
    '''
```

Comment (chrisniael) [reply powerline/segments/vim/plugin/coc.py]: Thanks for the tip.

**Comment by giovanniiapaolo** [comment]
I'd like to implement this in my setup, but I just can't seem to get it. I'm not even sure if it's available in the latest apt

**Pull request merged**